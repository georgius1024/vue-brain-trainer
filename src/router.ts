import Vue from 'vue';
import Router from 'vue-router';
import Home from './views/home.vue';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home,
    },
    {
      path: '/onboarding',
      name: 'onboarding',
      component: () => import(/* webpackChunkName: "onboarding" */ './views/onboarding.vue'),
    },
    {
      path: '/game',
      name: 'game',
      component: () => import(/* webpackChunkName: "game" */ './views/game.vue'),
    },
  ],
});
