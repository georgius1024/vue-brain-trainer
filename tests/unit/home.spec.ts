import { shallowMount } from '@vue/test-utils';
import Home from '@/views/home.vue';
import router from '@/router';
import store from '@/store';

describe('home.vue', () => {
  it('renders without errors', () => {
    const wrapper = shallowMount(Home, {
      router,
      store
    });
    expect(wrapper).toMatchSnapshot();
  });
  it('display training days', () => {
    const wrapper = shallowMount(Home, {
      router,
      store
    });
    expect(wrapper.html()).toContain((wrapper as any).vm.days(store.getters.daysInTraining));
  });
  it('display tasks stats', () => {
    const wrapper = shallowMount(Home, {
      router,
      store
    });
    store.commit('addTaskSolved');
    store.commit('addTasksFailed');
    expect(wrapper.html()).toContain(store.getters.tasksTotal);
    expect(wrapper.html()).toContain(store.state.tasksSolved);
    expect(wrapper.html()).toContain(store.getters.accuracy);
  });
  it('display duration and skill', () => {
    const wrapper = shallowMount(Home, {
      router,
      store
    });
    store.commit('changeDuration', 10);
    expect(wrapper.html()).toContain(store.state.duration);
    store.commit('changeSkill', 10);
    expect(wrapper.html()).toContain(store.state.skill);
  });
});
