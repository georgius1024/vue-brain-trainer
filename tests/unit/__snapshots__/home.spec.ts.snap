// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home.vue renders without errors 1`] = `
<div class="uk-container">
  <div class="uk-flex uk-flex-center">
    <div class="uk-card uk-card-default uk-card-hover uk-width-large">
      <div class="uk-card-header">
        <h3 class="uk-card-title">Привет!</h3>
      </div>
      <div class="uk-card-body">
        <p>Добро пожаловать на первый тренировочный день!</p>
        <p>Ваш последний результат: решено 0 из 0.</p>
        <p>Общая точность: 0%.</p>
        <h4>Настройки</h4>
        <form>
          <div class="uk-margin-top uk-margin-top">
            <div class="uk-align-left uk-margin-remove-bottom">1</div>
            <div class="uk-align-right uk-margin-remove-bottom">15</div><input type="range" min="1" max="15" step="1" class="uk-range">
            <div class="uk-flex uk-flex-center">Длительность: 7 минут</div>
          </div>
          <div class="uk-margin-top">
            <div class="uk-align-left uk-margin-remove-bottom">1</div>
            <div class="uk-align-right uk-margin-remove-bottom">10</div><input type="range" min="1" max="10" step="1" class="uk-range">
            <div class="uk-flex uk-flex-center">Сложность: 7</div>
          </div>
          <div class="uk-margin-top">
            <div><label><input type="checkbox" class="uk-checkbox"><span class="uk-margin-left">Сложение</span></label></div>
            <div><label><input type="checkbox" class="uk-checkbox"><span class="uk-margin-left">Вычитание</span></label></div>
            <div><label><input type="checkbox" class="uk-checkbox"><span class="uk-margin-left">Умножение</span></label></div>
            <div><label><input type="checkbox" class="uk-checkbox"><span class="uk-margin-left">Деление</span></label></div>
            <div><label><input type="checkbox" class="uk-checkbox"><span class="uk-margin-left">Возведение в степень</span></label></div>
          </div>
        </form>
      </div>
      <div class="uk-card-footer">
        <div class="uk-flex uk-flex-between uk-flex-middle uk-flex-row">
          <router-link-stub to="[object Object]" tag="a" event="click" class="uk-button uk-button-link"><span uk-icon="icon: question"></span><span class="uk-margin-small-left">Повторить обучение</span></router-link-stub>
          <router-link-stub to="[object Object]" tag="a" event="click" class="uk-button uk-button-default">Play!</router-link-stub>
        </div>
      </div>
    </div>
  </div>
</div>
`;
